src/Compilers/CSharp/Portable/BoundTree/DecisionTreeBuilder.cs

```diff
@@ -108,14 +105,11 @@ private DecisionTree AddByValue(DecisionTree.Guarded guarded, BoundConstantPatte
         {
             if (guarded.Default != null)
             {
-                if (guarded.Default.MatchIsComplete)
-                {
-                    return null;
-                }
+                Debug.Assert(!guarded.Default.MatchIsComplete); // otherwise we would have given a subsumption error
             }
             else
             {
-                guarded.Default = new DecisionTree.ByValue(guarded.Expression, guarded.Type, null);
+                guarded.Default = new DecisionTree.ByType(guarded.Expression, guarded.Type, null);
```

Comment:
Could you please capture that in a comment here?

---
In reply to: [100603286](https://github.com/dotnet/roslyn/pull/16928#discussion_r100603286) [](ancestors = 100603286,100600126,100598334,100474813)